### PR TITLE
Add feature to expose vars/defaults with include/import_role

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY
This PR aims to add functionality to `import_role` and `include_role` to allow exposing vars/defaults from the role.

`import_role` always exposes. `include_role` can be toggled, but defaults to `public=False`.

~This PR re-uses the `private` argument, but I debate flipping that to `public`, or at minimum renaming to a different argument as to cause less confusion.~

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
lib/ansible/playbook/helpers.py
lib/ansible/playbook/role_include.py

##### ANSIBLE VERSION

```
2.7
```

### CODE REVIEW

**Comment by sivel** [comment]
not_bot_skip_any_more

**Comment by sivel** [comment]
TODO:

* ~validate `vars_from` and how/where it occurs in regards to forking.~
  * tests show `vars/defaults_from` is working, these are loaded just like vars in `_load_role_data`

File: lib/ansible/playbook/helpers.py [hunk 1]
```diff
@@ -355,8 +355,7 @@ def load_list_of_tasks(ds, play, block=None, role=None, task_include=None, use_h
                         ir._role_name = templar.template(ir._role_name)
 
                     # uses compiled list from object
-                    blocks, _ = ir.get_block_list(variable_manager=variable_manager, loader=loader)
-                    task_list.extend(blocks)
+                    ir.get_block_list(variable_manager=variable_manager, loader=loader)
```
Comment (sivel) [inline lib/ansible/playbook/helpers.py]: Without this change, the tasks get executed twice.  Using this change causes the tasks of the role to execute before other tasks listed in `tasks`.

**Comment by samdoran** [comment]
One thing I'm wondering about is the timing of variable availability. Are role variables only available after the role has been called or immediately from the beginning of the play? I would think this is the case for `import_role` but maybe `included_rule` with `public: yes`, the vars are only avaiable after that role has been included since it's dynamic.

Also, are the vars available to subsequent plays in the same playbook?

I just thought these would be useful things to clarify in the final docs and possibly implement in the final code.

**Comment by sivel** [comment]
> One thing I'm wondering about is the timing of variable availability. Are role variables only available after the role has been called or immediately from the beginning of the play?

At the moment, there are tests describing this.  `import_role` exposes at parse time, `include_role` exposes at execution time.  So with `import_role` they are available before the tasks within execute, with `include_role` they only become available once the tasks within start executing.

Once, exposed, the same rules apply to roles vars that are defined under the `roles:` header.

**Comment by sivel** [comment]
I think this can come out of WIP now.

**Review by bcoca — APPROVED** [review]

**Comment by sivel** [comment]
(rebased)

**Comment by danihodovic** [comment]
This is awesome. My favorite feature of 2.7.0 Thanks @sivel

**Pull request merged**